### DISCUSSION TITLE
New version 2: a composed Reactour

### DISCUSSION THREAD

**Discussion by elrumordelaluz** [elrumordelaluz/reactour#373] [General]
Finally found that couple of days to finish (or restart/rewrite) lots of ideas I had for the next major version of `reactour`. I am sure I lost and forget a lot, but while writing the new library, another couple of features happens. 

The source of inspiration for all the new stuff was (an it is) all the _issues_ opened, while pointing bugs or similar, always lead to solutions and new features. So big thanks to all of the users of _Reactour_ who opens or simply used the library to port to a new version with new ideas and features. Also for the moment, I am sorry for all those features not implemented or forgotten and please keep in contact to report them.

### Changes

- Scoped package name under `@reactour`
- Monorepo
- Packages for specific use, consumable independiently
- Typescript codebase
- Switch from `styled-components` into `emotion`
- Flexible styles inspired by [react-select](https://react-select.com/styles)

<hr /><em>This discussion was created from the release <a href='https://github.com/elrumordelaluz/reactour/releases/tag/'>New version 2: a composed Reactour</a>.</em>

**Comment by came** [comment]
Good choice to use emotion :) However, I use it in my project as well ;) Now I get warnings telling me:

`You are loading @emotion/react when it is already loaded. Running multiple instances may cause problems. This can happen if multiple versions are used, or if multiple builds of the same version are used.`

Maybe you can make `emotion` a peer dependency in reactour as mui does?
https://mui.com/guides/styled-engine/

BR and thank you for the great components. I just got started ❤️

**Reply by elrumordelaluz** [reply]
absolutely! Could you please try `@reactour/tourr@2.3.0`?

**Reply by came** [reply]
wow ...you fixed it faster than I could check. Thanks a lot for the update. All is fine on my side :)
Thanks a lot
